src/ProposalTypesConfigurator.sol

```diff
@@ -164,12 +164,59 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
         if (!_proposalTypesExists[proposalTypeId]) revert InvalidProposalType();
 
         if (!scopeExists[proposalTypeId][txTypeHash]) revert InvalidScope();
-        Scope[] memory validScopes = scopes[proposalTypeId];
+        Scope memory validScope = scopes[proposalTypeId][txTypeHash];
+        return validScope.encodedLimits;
+    }
+
+    function getParameter(bytes calldata limit, uint256 startIdx, uint256 endIdx)
+        public
+        pure
+        returns (bytes memory parameter)
+    {
+        return limit[startIdx:endIdx + 1];
+    }
+
+    function validateProposedTx(bytes calldata proposedTx, uint8 proposalTypeId, bytes32 txTypeHash)
+        public
+        view
+        returns (bool valid)
+    {
+        Scope memory validScope = scopes[proposalTypeId][txTypeHash];
+        bytes memory scopeLimit = validScope.encodedLimits;
+        bytes4 selector = bytes4(scopeLimit);
```

Comment:
not sure i see we are checking that `encodedLimits` is at least bytes4. otherwise the casting to bytes4 could be unsafe.

asides from length, we should be checking that the first 4 bytes are not zero (i.e. they represent a function selector). requires some thinking on how this plays out when calling the fallback of a contract (does this version of proposal types support that? like sending ether)

Reply:
`parameters` variable inside the scope is `bytes` so it could easily lead to issues -- maybe it should be `bytes32`

Reply:
maybe it's fine to leave `parameters` as bytes so that the implementation is less complex, but we should have all needed checks

Reply:
I think we are going to not have support for fall back functions due to how we are managing scopes for each proposal type. For example since we need the type hashes on each of the proposal type objects